hypertrace-metrics-generator/hypertrace-metrics-generator/src/main/java/org/hypertrace/metrics/generator/MetricsProcessor.java

```diff
@@ -0,0 +1,162 @@
+import org.apache.kafka.streams.processor.PunctuationType;
+import org.apache.kafka.streams.processor.To;
+import org.apache.kafka.streams.state.KeyValueIterator;
+import org.apache.kafka.streams.state.KeyValueStore;
+import org.hypertrace.metrics.generator.api.Metric;
+import org.hypertrace.metrics.generator.api.MetricIdentity;
+import org.hypertrace.viewgenerator.api.RawServiceView;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class MetricsProcessor
+    implements Transformer<String, RawServiceView, KeyValue<byte[], ResourceMetrics>> {
+
+  private static final Logger logger = LoggerFactory.getLogger(MetricsProcessor.class);
+
+  private static final String TENANT_ID_ATTR = "tenant_id";
+  private static final String SERVICE_ID_ATTR = "service_id";
+  private static final String SERVICE_NAME_ATTR = "service_name";
+  private static final String API_ID = "api_id";
+  private static final String API_NAME = "api_name";
+  private static final String CONSUMER_ID_ATTR = "consumer_id";
+  private static final String METRIC_NUM_CALLS = "num_calls";
+  private static final String METRIC_NUM_CALLS_DESCRIPTION = "num of calls";
+  private static final String METRIC_NUM_CALLS_UNIT = "1";
+  private static final String DELIMITER = ":";
+  private static final String METRIC_AGGREGATION_TIME_MS = "metric.aggregation.timeMs";
+  private static final String METRIC_EMIT_WAIT_TIME_MS = "metric.emit.waitTimeMs";
+
+  private ProcessorContext context;
+  private KeyValueStore<MetricIdentity, Long> metricsIdentityStore;
+  private KeyValueStore<MetricIdentity, Metric> metricsStore;
+  private long metricAggregationTimeMs;
+  private long metricEmitWaitTimeMs;
+  private To outputTopicProducer;
+
+  @Override
+  public void init(ProcessorContext context) {
+
+    this.context = context;
+    this.metricsIdentityStore =
+        (KeyValueStore<MetricIdentity, Long>)
+            context.getStateStore(MetricsGenerator.METRICS_IDENTITY_STORE);
+    this.metricsStore =
+        (KeyValueStore<MetricIdentity, Metric>)
+            context.getStateStore(MetricsGenerator.METRICS_IDENTITY_VALUE_STORE);
+    this.outputTopicProducer = To.child(OUTPUT_TOPIC_PRODUCER);
+
+    Config jobConfig = (Config) (context.appConfigs().get(METRICS_GENERATOR_JOB_CONFIG));
+    this.metricAggregationTimeMs = jobConfig.getLong(METRIC_AGGREGATION_TIME_MS);
+    this.metricEmitWaitTimeMs = jobConfig.getLong(METRIC_EMIT_WAIT_TIME_MS);
+
+    restorePunctuators();
+  }
+
+  @Override
+  public KeyValue<byte[], ResourceMetrics> transform(String key, RawServiceView value) {
+    // construct metric attributes & metric
+    Map<String, String> attributes = new HashMap<>();
+    attributes.put(TENANT_ID_ATTR, value.getTenantId());
+    attributes.put(SERVICE_ID_ATTR, value.getServiceId());
+    attributes.put(SERVICE_NAME_ATTR, value.getServiceName());
+    attributes.put(API_ID, value.getApiId());
+    attributes.put(API_NAME, value.getApiName());
```

Comment:
we are skipping some columns from the view, like status_code etc, https://github.com/hypertrace/hypertrace-ingester/blob/main/hypertrace-view-generator/hypertrace-view-creator/src/main/resources/configs/raw-service-view/application.conf#L21
how will filtering work on those attributes?

Reply:
Adding status_code and protocol all string-related attributes other than IDs.

Reply:
@rish691 I had included all the attributes which one mapped in SERVICE and API entities - https://github.com/hypertrace/query-service/blob/main/query-service/src/main/resources/configs/common/application.conf#L97

https://github.com/hypertrace/query-service/blob/main/query-service/src/main/resources/configs/common/application.conf#L275

However, protocol, and status_code make sense, so extended it.

Reply:
Done